lib/block-supports/typography.php

```diff
@@ -455,22 +458,35 @@ function gutenberg_get_typography_font_size_value( $preset, $should_use_fluid_ty
 		return $preset['size'];
 	}
 
-	// Checks if fluid font sizes are activated.
-	$global_settings     = gutenberg_get_global_settings();
-	$typography_settings = isset( $global_settings['typography'] ) ? $global_settings['typography'] : array();
-	$layout_settings     = isset( $global_settings['layout'] ) ? $global_settings['layout'] : array();
+	/*
+	 * Backwards compatibility since 6.5.
+	 * As a bool (deprecated since 6.5), $settings acts as an override to switch fluid typography "on" (`true`) or "off" (`false`).
+	 */
+	if ( is_bool( $settings ) ) {
+		_deprecated_argument( __FUNCTION__, '6.5.0', __( '`boolean` type for second argument `$settings` is deprecated. Use `array()` instead.', 'gutenberg' ) );
+		$settings = array(
+			'typography' => array(
+				'fluid' => $settings,
+			),
+		);
+	}
 
-	$should_use_fluid_typography
-		= isset( $typography_settings['fluid'] ) &&
-		( true === $typography_settings['fluid'] || is_array( $typography_settings['fluid'] ) ) ?
-		true :
-		$should_use_fluid_typography;
+	// Fallback to global settings as default.
+	$global_settings             = gutenberg_get_global_settings();
+	$settings                    = wp_parse_args(
+		$settings,
+		$global_settings
+	);
+	$typography_settings         = isset( $settings['typography'] ) ? $settings['typography'] : array();
+	$should_use_fluid_typography = isset( $typography_settings['fluid'] ) && ! empty( $typography_settings['fluid'] );
```

Comment:
Tiny nit: do we still need the `isset()` half of this check since we have `! empty()` as the second half of it? I.e. would the following be equivalent?

```suggestion
	$should_use_fluid_typography = ! empty( $typography_settings['fluid'] );
```